From 240192f553cc71ddbbe29d8242f7c7d956b17202 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 31 Aug 2017 14:17:52 +0800
Subject: [PATCH 2473/5242] MLK-16349 video: fbdev: dcss: reset dcss domains
 before clock source config

commit  44789899c284863f01194f1a87055d71f99b120c from
https://source.codeaurora.org/external/imx/linux-imx.git

Before accssing any DCSS register, add the DCSS domains
reset before DCSS clock source configuration, since DCSS
clock source configuration requires all the DCSS domains
to be in reset state according to the DCSS specification.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 8ae9186..caca20c 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -3176,7 +3176,8 @@ static int dcss_probe(struct platform_device *pdev)
 	if (ret)
 		goto kfree_info;
 
-	/* TODO: reset DCSS to make it clean */
+	/* reset DCSS to make sure in reset state*/
+	writel(0xffffffff, info->blkctl_base + 0x8);
 
 	/* Clocks select: before dcss de-resets */
 	if (!strcmp(info->disp_dev, "hdmi_disp"))
-- 
1.7.9.5

